### DISCUSSION TITLE
custom field saving strategy

### DISCUSSION THREAD

**Discussion by jdion84** [laravel/nova-issues#5850] [Q&A] [answered]
i am trying to set up EAV for a model called `Lead`, basically allowing end-users to create custom fields for leads

the `Lead` model contains a `hasMany` relation to `CustomField` model

is there a way i can specify the saving strategy so that it looks like a regular nova field, but actually uses relation in the background?

for example:

```php
$customFieldId = 3;

Text::make("Custom Field $customFieldId", "customFields.$customFieldId")
    ->saveUsing(function ($request, $model, $attribute) {
        $id = Str::replaceFirst('customFields.', '', $attribute);

        $model->customFields()->updateOrCreate([
            'id' => $id,
        ], [
            'value' => $request->input($attribute),
        ]);
    });
```

**Comment by jdion84** [comment]
i know i can use JSON field in `Lead` model then an observer to sync the actual relations, but was wondering if theres a native way to do via nova

**Reply by treed-eaglemetal** [reply, marked as answer]
It sounds like you're looking for `fillUsing`.

https://nova.laravel.com/docs/resources/fields.html#field-hydration

If you need something to happen after the model is saved, you can return a closure.
